docsrc/allblocks.nim

```diff
@@ -1,40 +1,126 @@
 import nimoji
 import nimib
+import nimpy
+import std/[math, strutils, strformat]
 
 nbInit
 
+var nbToc: NbBlock
+
+template addToc =
+  newNbBlock("nbText", false, nb, nbToc, ""):
+    nbToc.output = "## Table of Contents:\n\n"
+
+template nbCodeBlock(name:string) =
+  let anchorName = name.toLower.replace(" ", "-")
+  nbText "<a name = \"" & anchorName & "\"></a>\n### " & name & "\n\n---"
+  # see below, but any number works for a numbered list
+  nbToc.output.add "1. <a href=\"#" & anchorName & "\">" & name & "</a>\n"
+
 nbText: """
 > This nimib document provides a brief description and example for 
 > all blocks in nimib.
 
 :warning: **This document is not finished as of now.**
 """.emojize
 
-nbText: """
-### nbCodeSkip
+addToc()
 
+nbCodeBlock:"nbCodeSkip"
+nbText: """
 Similar to `nbCode`, `nbCodeSkip` is a block that displays 
 highlighted code but does not compile or run it.
```

Comment:
Since we (me and Pietro) use the fact that nimib can garantuee that your code compiles as one of the big arguments for using nimib, I think it would be worth it to add a big disclaimer on `nbCodeSkip` that makes that **very** clear. Something like:
```md
> `nbCodeSkip` does **not** garantuee in any way that the code that you show will compile nor run.
```

Reply:
I am not sure I feel **that** strongly about this 😉, but more emphasis would not hurt :)

Reply:
I used a ⚠️ emoji for this.